packages/grid/x-data-grid/src/internals/models/api/gridColumnApi.ts

```diff
@@ -47,12 +47,12 @@ export interface GridColumnApi {
    * @param {GridColDef} col The new [[GridColDef]] object.
    * @deprecated Use `apiRef.current.updateColumns` instead.
    */
-  updateColumn: (col: GridColDef) => void;
+  updateColumn: <Api extends GridApiCommon = GridApiCommon>(col: GridColDef<Api>) => void;
   /**
    * Updates the definition of multiple columns at the same time.
    * @param {GridColDef[]} cols The new column [[GridColDef]] objects.
    */
-  updateColumns: (cols: GridColDef[]) => void;
+  updateColumns: <Api extends GridApiCommon = GridApiCommon>(cols: GridColDef<Api>[]) => void;
```

Comment:
For column methods I added the API as a generic because I can't easily now which version of `GridApi` is used from inside one of `GridApi` methods.

Reply:
Not sure to understand your poin here.
If the method does not have the generic to pass it to the interface, how does the user override it ?

Reply:
We are repeating `<Api extends GridApiCommon = GridApiCommon>` in almost every method. I was hoping that we could move the generic to the interface but it didn't work. Even passing the correct API in `GridApiPro` it goes back to `GridApiCommunity`. We'll be able to simplify the types once we remove `api` from the params.

```diff
diff --git a/packages/grid/x-data-grid/src/internals/models/api/gridColumnApi.ts b/packages/grid/x-data-grid/src/internals/models/api/gridColumnApi.ts
index 8865f822c..d51a02a8c 100644
--- a/packages/grid/x-data-grid/src/internals/models/api/gridColumnApi.ts
+++ b/packages/grid/x-data-grid/src/internals/models/api/gridColumnApi.ts
@@ -6,23 +6,23 @@ import type { GridApiCommon } from './gridApiCommon';
  * The column API interface that is available in the grid [[apiRef]].
  * TODO: Differentiate interfaces based on the plan
  */
-export interface GridColumnApi {
+export interface GridColumnApi<Api extends GridApiCommon = GridApiCommon> {
   /**
    * Returns the [[GridColDef]] for the given `field`.
    * @param {string} field The column field.
    * @returns {{GridStateColDef}} The [[GridStateColDef]].
    */
-  getColumn: <Api extends GridApiCommon = GridApiCommon>(field: string) => GridStateColDef<Api>;
+  getColumn: (field: string) => GridStateColDef<Api>;
```

Reply:
Yes, this generic is a pain.
I think we could stop typing the `api` in the params once it's deprecated.